src/interceptors/ClientRequest/createClientRequestOverride.ts

```diff
@@ -121,9 +121,9 @@ export function createClientRequestOverride(
         }
       }
 
-      setImmediate(() => {
+      setTimeout(() => {
         this.emit('drain')
-      })
+      }, 0)
```

Comment:
`0` is the default value according the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#parameters):

> If this parameter is omitted, a value of 0 is used, meaning execute "immediately", or more accurately, the next event cycle. 


This means you can omit the argument for the delay.
```suggestion
      })
```

Reply:
Yeah, I thought using `0` as timeout made it more clear (without reading the dosc) that it will immediately run

Reply:
Not sure which I prefer, personally I lean towards being explicit about the intention i.e. what weyert is suggesting, that way the intention is self documented as opposed to leaving it open to the possibility of someone assuming `0` was forgotten to be passed.

Either way I don't mind, just thought I'd come by and thank you guys for doing a fix, been bugging me all morning :)

Reply:
Explicit `0` is perfectly fine, you can leave it like it is.

Reply:
Sweet :)